
package codec;

message MsgVersion {
	required uint64 proto_ver = 1;
	required string client_ver = 3;
	required uint64 node_id = 2;
}

message MsgPingPong {
	required uint64 cookie = 1;
}

message Address {
	required uint64 proto_ver = 1;
	required uint64 time = 2;
	required uint64 flags = 3;
	required string ip = 4;
	required uint32 port = 5;
}

message MsgAddresses {
	repeated Address peers = 1;
}

message MsgDHTMisc {
	required uint64 request_id = 1;
}

message MsgDHTKey {
	required uint64 request_id = 1;
	required bytes key = 2;
	optional uint64 flags = 3 [default = 0];
}

message MsgDHTKeyValue {
	required uint64 request_id = 1;
	required bytes key = 2;
	required bytes value = 3;
	optional uint64 flags = 4 [default = 0];
}

message MsgDHTNode {
	required uint64 node_id = 1;
	required string ip = 2;
	required uint32 port = 3;
	optional uint64 flags = 4 [default = 0];
}

message MsgDHTNodes {
	required uint64 request_id = 1;
	repeated MsgDHTNode nodes = 2;
}

